README.md

```diff
@@ -145,7 +189,10 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+sentence.cpp
+To run in terminal: 
+g++ -std=c++11 sentence.cpp
+./a.out
```

Comment:
You used a hash table with is great. But your testing wasn't really testing: it just output stuff about sentences. An automated test should know the expected result and make it clear when the function being tested fails to produce it.

Also, a bunch of people wrote code like this:

~~~ cpp
    vector<string> s1 = { "the", "cat", "ate", "EOI" };
    vector<string> s2 = {"the", "dog", "ran", "slowly", "EOI"};
    vector<string> s3 = {"the", "lazy", "cat", "ran", "slowly", "EOI"};
    vector<string> s4 = {"the", "smelly", "dog", "ran", "EOI"};
    vector<string> s5 = {"smelly", "dog", "ran", "EOI"};
    vector<string> s6 = {"the", "cat", "ate"};
    vector<string> s7 = {"the", "cat", "EOI"};
    vector<string> s8 = {"the", "smelly", "lazy", "cat", "ate", "EOI"};

    vector<vector<string> > sentences;

    sentences.push_back(s1);
    sentences.push_back(s2);
    sentences.push_back(s3);
    sentences.push_back(s4);
    sentences.push_back(s5);
    sentences.push_back(s6);
    sentences.push_back(s7);
    sentences.push_back(s8);
~~~

I don't get it. C++ is already verbose enough without adding to your woes. How about:

~~~ cpp
    vector<vector<string>> sentences {
        { "the", "cat", "ate", "EOI" },
        {"the", "dog", "ran", "slowly", "EOI"},
        {"the", "lazy", "cat", "ran", "slowly", "EOI"},
        {"the", "smelly", "dog", "ran", "EOI"},
        {"smelly", "dog", "ran", "EOI"},
        {"the", "cat", "ate"},
        {"the", "cat", "EOI"},
        {"the", "smelly", "lazy", "cat", "ate", "EOI"}
    };
~~~

(No deduction for that, but -2 for the tests)

# -2